core/src/impl/Kokkos_Core.cpp

```diff
@@ -613,7 +613,7 @@ void pre_initialize_internal(const Kokkos::InitializationSettings& settings) {
 #endif
 
   declare_configuration_metadata("architecture", "Default Device",
-                                 typeid(Kokkos::DefaultExecutionSpace).name());
+                                 Kokkos::DefaultExecutionSpace::name());
```

Comment:
Just surprised that there is no test for that.

Could be a simple regex search in the output.

Reply:
There is no test for it currently.  The 1st "incremental" test calls `ExecSpace::print_configuration()` directly but does not assert anything.
We can pursue this elsewhere, this change stands on its own.